challenge.c

```diff
@@ -100,8 +134,17 @@ int32_t q4(uint8_t * array, uint32_t arrayLength) {
 // q9(&x, &y); // returns 0
 // Now, x = 10 and y = 5
 //-------------------------------------------------------------------------
+
 int q9(int *a, int *b) {
+    if (!a || !b) {
+        return -1;
+    };
+
+    *a ^= *b;
```

Comment:
Although this works, using a temp variable to perform the swap is a more readable solution (it clearly shows the intent of the code). In general, readability is preferred over minor optimizations.